File: Entregable 2.R
# Cargar librerías necesarias
library(ggplot2)

# Generación de datos simulados
set.seed(42)
kilometraje <- runif(100, 10000, 100000)  # Generación de 100 observaciones entre 10,000 y 100,000 km
error <- rnorm(100, mean = 0, sd = 2000)  # Error con media 0 y desviación estándar de 2000
precio <- 50000 - 0.5 * kilometraje + error  # Cálculo del precio con la fórmula dada

# Crear un DataFrame
data <- data.frame(Kilometraje = kilometraje, Precio = precio)

# Ajustar el modelo de regresión lineal
modelo <- lm(Precio ~ Kilometraje, data = data)

# Ver el resumen del modelo
summary(modelo)

# Coeficientes del modelo
coef(modelo)

# Gráfico de dispersión con la línea de regresión
ggplot(data, aes(x = Kilometraje, y = Precio)) +
  geom_point(color = 'blue') +
  geom_smooth(method = 'lm', color = 'red') +
  labs(title = 'Regresión Lineal: Relación entre Kilometraje y Precio', 
       x = 'Kilometraje (km)', 
       y = 'Precio ($)') +
  theme_minimal()

# Predicciones para kilometrajes específicos
kilometrajes_nuevos <- c(20000, 50000, 80000)
precios_predichos <- predict(modelo, newdata = data.frame(Kilometraje = kilometrajes_nuevos))

# Imprimir predicciones
for (i in 1:length(kilometrajes_nuevos)) {
  cat(sprintf("El precio predicho para un automóvil con %d km es: $%.2f\n", kilometrajes_nuevos[i], precios_predichos[i]))
}

File: entregable2.py
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score

# Generación de datos simulados
np.random.seed(42)
kilometraje = np.random.uniform(10000, 100000, 100)  # Generación de 100 observaciones entre 10,000 y 100,000 km
error = np.random.normal(0, 2000, 100)  # Error con media 0 y desviación estándar de 2000
precio = 50000 - 0.5 * kilometraje + error  # Cálculo del precio con la fórmula dada

# Creación de un DataFrame para almacenar los datos
data = pd.DataFrame({'Kilometraje': kilometraje, 'Precio': precio})

# Separación de las variables predictoras y la variable objetivo
X = data[['Kilometraje']]
y = data['Precio']

# Creación y ajuste del modelo de regresión lineal
modelo = LinearRegression()
modelo.fit(X, y)

# Obtención de los coeficientes y el valor R²
print(f"Coeficiente de regresión (pendiente): {modelo.coef_[0]:.4f}")
print(f"Intercepción: {modelo.intercept_:.2f}")
print(f"Valor de R²: {modelo.score(X, y):.4f}")

# Creación del gráfico de dispersión con la línea de regresión
plt.figure(figsize=(10, 6))
plt.scatter(data['Kilometraje'], data['Precio'], color='blue', label='Datos observados')
plt.plot(data['Kilometraje'], modelo.predict(X), color='red', label='Línea de regresión')
plt.xlabel('Kilometraje (km)')
plt.ylabel('Precio ($)')
plt.title('Regresión Lineal: Relación entre Kilometraje y Precio')
plt.legend()
plt.grid(True)
plt.show()

# Predicciones para kilometrajes específicos
kilometrajes_nuevos = np.array([20000, 50000, 80000]).reshape(-1, 1)
precios_predichos = modelo.predict(kilometrajes_nuevos)

# Impresión de las predicciones
for km, precio in zip(kilometrajes_nuevos, precios_predichos):
    print(f"El precio predicho para un automóvil con {km[0]} km es: ${precio:.2f}")